src/util/Date.h

```diff
@@ -303,6 +303,12 @@ class Date {
     timeZone_ = static_cast<unsigned>(actualTimeZone - minTimeZoneActually);
   }
 
+  // True iff Date object has set time values (hours, minutes, seconds) (i.e
+  //  "This is a `xsd:dateTime`")
+  [[nodiscard]] bool hasTime() const {
+    return getHour() != -1;
+  }
+
```

Comment:
That you very much. When you introduce the helper, you can also consistently apply it to existing code (where it can be easily spotted). In the Date.cpp there are two occurences of `getHour() == -1` that can be replaced by this function.

Reply:
Here I only found one occurrence. The only second contender would be `Date.cpp:179`, but since there is no date object yet created I cannot use the `hasTime()` function. Was that the second occurrence or did i overlook sth?